l10n_it_intrastat/models/account.py

```diff
@@ -143,16 +143,27 @@ def _prepare_intrastat_line_province_origin(self, company_id, res):
 
     def _prepare_intrastat_line_country_good_origin(self, res):
         self.ensure_one()
-        country_good_origin_id = self.env["res.country"].browse()
-        if self.move_id.is_sale_document():
+        if self.mapped("product_id.product_tmpl_id.intrastat_country_origin_id"):
+            country_good_origin_id = (
+                self.product_id.product_tmpl_id.intrastat_country_origin_id
+            )
+        elif self.mapped(
+            "product_id.product_tmpl_id.categ_id.intrastat_country_origin_id"
+        ):
+            country_good_origin_id = (
+                self.product_id.product_tmpl_id.categ_id.intrastat_country_origin_id
+            )
+        elif self.move_id.is_sale_document():
             country_good_origin_id = self.move_id.company_id.partner_id.country_id
         elif self.move_id.is_purchase_document():
             country_good_origin_id = self.move_id.partner_id.country_id
+        else:
+            country_good_origin_id = self.env["res.country"]
```

Comment:
Prima invece di code-review ho erroneamente lasciato un commento generico: https://github.com/OCA/l10n-italy/pull/3256#pullrequestreview-1668210873 sono questi i checks di cui chiedevo conferma, se è giusto che rimangano solo qui o se sono da da portare anche in in [_prepare_intrastat_line_country_origin](https://github.com/OCA/l10n-italy/blob/2d4b4542e61b4378d66514863e0a763329f51c43/l10n_it_intrastat/models/account.py#L153C56-L153C56)

Reply:
è giusto che il controllo sia solo in good origin che indica il paese di origine della merce l'altra funzione e per la destinazione